S3StorageWagon/src/main/java/com/gkatzioura/maven/cloud/s3/CredentialsFactory.java

```diff
@@ -19,17 +19,23 @@
 import org.apache.maven.wagon.authentication.AuthenticationInfo;
 
 import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.AWSCredentialsProviderChain;
 import com.amazonaws.auth.AWSStaticCredentialsProvider;
 import com.amazonaws.auth.BasicAWSCredentials;
 import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
 
 public class CredentialsFactory {
 
     public AWSCredentialsProvider create(AuthenticationInfo authenticationInfo) {
-        if(authenticationInfo==null) {
-            return new DefaultAWSCredentialsProviderChain();
+        if(authenticationInfo != null && authenticationInfo.getUserName() != null && authenticationInfo.getPassword() != null) {
```

Comment:
authenticationInfo remained non-null even when I didn't provide a username or password (as in my example), so I added additional checks, and also created a chained fallback from using those credentials to the default chain.